server/room.ts

```diff
@@ -66,353 +67,37 @@ export class Room {
-            return;
-          }
-          if (!this.validateLock(socket.id)) {
-            return;
-          }
-          if (!data) {
-            return;
-          }
-          this.isAssigningVM = true;
-          let isLarge = false;
-          if (process.env.STRIPE_SECRET_KEY && data && data.uid && data.token) {
-            const decoded = await validateUserToken(data.uid, data.token);
-            // Check if user is subscriber, if so allow isLarge
-            if (decoded?.email) {
-              const customer = await getCustomerByEmail(decoded.email);
-              if (customer?.subscriptions?.data?.[0]?.status === 'active') {
-                console.log('found active sub for ', customer?.email);
-                isLarge = data.options?.size === 'large';
-              }
-            }
-          }
-
-          if (process.env.RECAPTCHA_SECRET_KEY) {
-            try {
-              // Validate the request isn't spam/automated
-              const validation = await axios({
-                url: `https://www.google.com/recaptcha/api/siteverify?secret=${process.env.RECAPTCHA_SECRET_KEY}&response=${data.rcToken}`,
-                method: 'POST',
-              });
-              console.log(validation?.data);
-              const isLowScore = validation?.data?.score < 0.12;
-              const failed = validation?.data?.success === false;
-              if (failed || isLowScore) {
-                if (isLowScore) {
-                  redisCount('recaptchaRejectsLowScore');
-                } else {
-                  redisCount('recaptchaRejectsOther');
-                }
-                return;
-              }
-            } catch (e) {
-              // if Recaptcha is down or other network issues, allow continuing
-              console.warn(e);
-            }
-          }
-
-          redisCount('vBrowserStarts');
-          this.cmdHost(socket, 'vbrowser://');
-          const vmManager = isLarge
-            ? this.vmManagers?.large
-            : this.vmManagers?.standard;
-          const assignment = await vmManager?.assignVM();
-          if (!this.isAssigningVM) {
-            // Maybe the user cancelled the request before assignment finished
-            return;
-          }
-          this.isAssigningVM = false;
-          if (!assignment) {
-            this.cmdHost(socket, '');
-            this.vBrowser = undefined;
-            return;
-          }
-          this.vBrowser = assignment;
-          this.roster.forEach((user, i) => {
-            if (user.id === socket.id) {
-              this.roster[i].isController = true;
-            } else {
-              this.roster[i].isController = false;
-            }
-          });
-          this.cmdHost(
-            undefined,
-            'vbrowser://' + this.vBrowser.pass + '@' + this.vBrowser.host
-          );
-          io.of(roomId).emit('roster', this.roster);
-        }
+      socket.on('CMD:name', (data) => this.changeUserName(data, socket));
```

Comment:
maybe we should make these socket, data since socket is always passed but data is optional

Reply:
socket as the first arg? sure thing!